packages/machine-extractor/src/toMachineConfig.ts

```diff
@@ -199,28 +202,25 @@ export const getActionConfig = (
   const actions: Actions<any, any> = [];
 
   astActions?.forEach((action) => {
-    if (
-      opts?.anonymizeInlineImplementations &&
-      action.declarationType !== 'named'
-    ) {
-      actions.push({
-        type: 'anonymous',
-      });
-    } else if (
-      opts?.hashInlineImplementations &&
-      action.declarationType !== 'named'
-    ) {
-      actions.push({
-        type: action.inlineDeclarationId,
-      });
-    } else {
-      if (opts?.stringifyInlineImplementations) {
+    switch (true) {
+      case action.declarationType === 'named':
+        actions.push(action.name);
+        return;
+      case opts?.anonymizeInlineImplementations:
+        actions.push({
+          type: 'anonymous',
+        });
+        return;
+      case opts?.hashInlineImplementations:
+        actions.push({
+          type: action.inlineDeclarationId,
+        });
+        return;
+      case opts?.stringifyInlineImplementations:
         actions.push(
```

Comment:
Should this be guarded, just in case?

```ts
if (opts && action.node.start && action.node.end) {
  actions.push(
    opts.fileContent.slice(action.node.start, action.node.end),
  );
}
```

Reply:
Not really - those properties are truly always there in our case. The types just don't reflect that as they are too close to what Babel exposes. I plan to improve this in the future but since they are always there it would obscure the code a lot if we'd guard against this all over the place.